packages/miniflare/src/workers/kv/assets.worker.ts

```diff
@@ -13,21 +17,26 @@ export default <ExportedHandler<Env>>{
 			return new Response(message, { status: 405, statusText: message });
 		}
 
-		// Decode key
-		const url = new URL(request.url);
-		let key = url.pathname.substring(1); // Strip leading "/"
+		const decoder = new TextDecoder();
+		const reverseMap: AssetReverseMap = JSON.parse(
+			decoder.decode(env.__STATIC_ASSETS_REVERSE_MAP)
+		);
 
-		if (url.searchParams.get(KVParams.URL_ENCODED)?.toLowerCase() === "true") {
-			key = decodeURIComponent(key);
-		}
+		// don't uri decode pathname, because we encode the filepath before hashing
+		const key = new URL(request.url).pathname.substring(1);
+
+		const entry = reverseMap[key] ?? {};
+		const filePath = entry["filePath"] ?? "";
+		// falls back to application/octet-stream
+		// const contentType = entry["contentType"];
 
 		const blobsService = env[SharedBindings.MAYBE_SERVICE_BLOBS];
-		if (key === "" || key === "/") {
+		if (filePath === "" || filePath === "/") {
```

Comment:
```suggestion
		if (filePath === undefined || filePath === "/") {
```

Reply:
I would expect `""` to be an invalid entry in the reverseMap ...
Also, when would the reverseMap contain `"/"`?

Reply:
Yeah I thought that was a bit funny, should've flagged it - before i remove it, @CarmenPopoviciu was there a particular reason this was checking for "/"?

Reply:
this from before any sort of assetManifest impl. days, and is prolly not relevant anymore